src/legacy/core_plugins/kibana/public/visualize/np_ready/editor/editor.js

```diff
@@ -388,22 +403,16 @@ function VisualizeAppController(
 
     const initialState = stateContainer.getState();
 
-    $scope.appState = {
-      // mock implementation of the legacy appState.save()
-      // this could be even replaced by passing only "updateAppState" callback
-      save() {
-        stateContainer.transitions.updateVisState(vis.getState());
-      },
-    };
-
     const handleLinkedSearch = linked => {
       if (linked && !savedVis.savedSearchId && savedSearch) {
         savedVis.savedSearchId = savedSearch.id;
-        vis.savedSearchId = savedSearch.id;
+        vis.data.savedSearchId = savedSearch.id;
         searchSource.setParent(savedSearch.searchSource);
+        updateEditor.next();
```

Comment:
You'll avoid calling these functions, if you revert passing down the `isLinked` flag.